Property 'yaml' does not exist on type 'typeof languages'

i'm using `ngx-monaco-editor` (https://github.com/atularen/ngx-monaco-editor#readme) library in my Angular project. also i would like to use `monaco-yaml`: 

`monaco.languages.yaml.yamlDefaults.setDiagnosticsOptions({ ... })`

for some reason i get following error:

`error TS2339: Property 'yaml' does not exist on type 'typeof languages'.`

i cannot understand, what's the problem: `ngx-monaco-editor` already contains `monaco-editor` lib, and `monaco.languages` is defined, but there is no `yaml` property. that's kinda strange for me, since in `monaco-yaml` is declared same `monaco` namespace with property `yaml`, so it's global scope, but my code see only `monaco` namespase properties defined in `ngx-monaco-editor`

i'll be happy if anyone has any ideas about what could that be

--- a/src/monaco.d.ts
+++ b/src/monaco.d.ts
@@ -2,53 +2,56 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
+import { IEvent } from 'monaco-editor';
 
-declare namespace monaco.languages.yaml {
-  export interface DiagnosticsOptions {
-    /**
-     * If set, the validator will be enabled and perform syntax validation as well as schema based validation.
-     */
-    readonly validate?: boolean;
-
-    /**
-     * A list of known schemas and/or associations of schemas to file names.
-     */
-    readonly schemas?: Array<{
+declare module 'monaco-editor' {
+  namespace languages.yaml {
+    export interface DiagnosticsOptions {
       /**
-       * The URI of the schema, which is also the identifier of the schema.
+       * If set, the validator will be enabled and perform syntax validation as well as schema based validation.
        */
-      readonly uri: string;
-      /**
-       * A list of file names that are associated to the schema. The '*' wildcard can be used. For example '*.schema.json', 'package.json'
-       */
-      readonly fileMatch?: string[];
-      /**
-       * The schema for the given URI.
-       */
-      readonly schema?: any;
-    }>;
+      readonly validate?: boolean;
 
-    /**
-     *  If set, the schema service would load schema content on-demand with 'fetch' if available
-     */
-    readonly enableSchemaRequest?: boolean;
-    /**
-     * If specified, this prefix will be added to all on demand schema requests
-     */
-    readonly prefix?: string;
-    /**
-     * Whether or not kubernetes yaml is supported
-     */
-    readonly isKubernetes?: boolean;
+      /**
+       * A list of known schemas and/or associations of schemas to file names.
+       */
+      readonly schemas?: Array<{
+        /**
+         * The URI of the schema, which is also the identifier of the schema.
+         */
+        readonly uri: string;
+        /**
+         * A list of file names that are associated to the schema. The '*' wildcard can be used. For example '*.schema.json', 'package.json'
+         */
+        readonly fileMatch?: string[];
+        /**
+         * The schema for the given URI.
+         */
+        readonly schema?: any;
+      }>;
 
-    readonly format?: boolean;
+      /**
+       *  If set, the schema service would load schema content on-demand with 'fetch' if available
+       */
+      readonly enableSchemaRequest?: boolean;
+      /**
+       * If specified, this prefix will be added to all on demand schema requests
+       */
+      readonly prefix?: string;
+      /**
+       * Whether or not kubernetes yaml is supported
+       */
+      readonly isKubernetes?: boolean;
+
+      readonly format?: boolean;
+    }
+
+    export interface LanguageServiceDefaults {
+      readonly onDidChange: IEvent<LanguageServiceDefaults>;
+      readonly diagnosticsOptions: DiagnosticsOptions;
+      setDiagnosticsOptions(options: DiagnosticsOptions): void;
+    }
+
+    export const yamlDefaults: LanguageServiceDefaults;
   }
-
-  export interface LanguageServiceDefaults {
-    readonly onDidChange: IEvent<LanguageServiceDefaults>;
-    readonly diagnosticsOptions: DiagnosticsOptions;
-    setDiagnosticsOptions(options: DiagnosticsOptions): void;
-  }
-
-  export const yamlDefaults: LanguageServiceDefaults;
 }

